src/components/views/rooms/EventTile.tsx

```diff
@@ -496,22 +496,11 @@ export class UnwrappedEventTile extends React.Component<EventTileProps, IState>
     };
 
     private get thread(): Thread | null {
-        if (!SettingsStore.getValue("feature_thread")) {
+        if (!SettingsStore.getValue("feature_threadstable")) {
             return null;
         }
 
-        let thread = this.props.mxEvent.getThread();
-        /**
-         * Accessing the threads value through the room due to a race condition
-         * that will be solved when there are proper backend support for threads
-         * We currently have no reliable way to discover than an event is a thread
-         * when we are at the sync stage
-         */
-        if (!thread) {
-            const room = MatrixClientPeg.get().getRoom(this.props.mxEvent.getRoomId());
-            thread = room?.findThreadForEvent(this.props.mxEvent);
-        }
-        return thread ?? null;
+        return this.props.mxEvent.getThread() ?? null;
```

Comment:
Will this not re-introduce the race condition if the user's HS doesn't support the backend support yet, e.g. Dendrite?

Reply:
I did not consider dendrite in the mix, good shout 👍

Reply:
No need it be even Dendrite, we often get rageshakes from servers without the /hierarchy API, those are over a year old